onnxruntime/core/session/inference_session.cc

```diff
@@ -351,17 +354,46 @@ void InferenceSession::SetLoggingManager(const SessionOptions& session_options,
 void InferenceSession::ConstructorCommon(const SessionOptions& session_options,
                                          const Environment& session_env) {
   auto status = FinalizeSessionOptions(session_options, model_proto_, is_model_proto_parsed_, session_options_);
-  // a monotonically increasing session id for use in telemetry
-  session_id_ = global_session_id_.fetch_add(1);
   ORT_ENFORCE(status.IsOK(), "Could not finalize session options while constructing the inference session. Error Message: ",
               status.ErrorMessage());
 
+  // a monotonically increasing session id for use in telemetry
+  session_id_ = global_session_id_.fetch_add(1);
+  std::lock_guard<OrtMutex> lock(active_sessions_mutex_);
```

Comment:
Move the code to "#ifdef _WIN32"? It needs to acquire a giant lock and provides no benefit to Linux users.

Reply:
Sure, I can change this to be Win32 for until something else cross-platform needs to look at active sessions. Although session creation/destruction should not be a hot code path.